Automattic/sensei
# Add filters to hide users, move some common code to Temporary users c…

Fixes #6386 #6394

### Changes proposed in this Pull Request

* Apply filters to both guest and preview users
* Add more filters to hide these users from the Students and Grading screens too
  * Still show a preview/guest user that has an 'Ungraded' quiz, to support manually graded quizzes
* Always load the admin filters so REST API is filtered too
* Move shared code around to the `Sensei_Temporary_User` class
* Also remove preview users after a week, rename data cleaner class

### Testing instructions

* Create an open access course and take it as a guest student
* Also activate student preview as a teacher for a course and complete a few lessons
* Check that the users and roles are not showing up on the following admin screens:
	* Users → All Users
	* Sensei → Grading
	* Sensei → Students
	* Sensei → Reports
	* Sensei → Groups → [Some group] → Add students
	* On the Sensei screens, also check filtering the pages by course/lesson, status, and searching for users


### New/Updated Hooks

`sensei_learners_query`: Filter the query to list learners on the 'Students' screen

'sensei_check_for_activity': Added a second arguments with the original query arguments.
